watertap/tools/parameter_sweep.py

```diff
@@ -319,108 +338,239 @@ def _interp_nan_values(global_values, global_results):
 
 # ================================================================
 
-def parameter_sweep(model, sweep_params, outputs, results_file=None, optimize_function=_default_optimize,
-        optimize_kwargs=None, reinitialize_function=None, reinitialize_kwargs=None,
-        mpi_comm=None, debugging_data_dir=None, interpolate_nan_outputs=False, num_samples=None, seed=None):
+def _create_local_output_skeleton(model, sweep_params, outputs, num_samples):
 
-    '''
-    This function offers a general way to perform repeated optimizations
-    of a model for the purposes of exploring a parameter space while
-    monitoring multiple outputs. 
-    If provided, writes single CSV file to ``results_file`` with all inputs and resulting outputs.
+    output_dict = {}
+    output_dict["sweep_params"] = {}
+    output_dict["outputs"] = {}
 
-    Arguments:
+    var_str_list = []
 
-        model : A Pyomo ConcreteModel containing a watertap flowsheet, for best 
-                results it should be initialized before being passed to this
-                function.
+    # Store the inputs
+    for key in sweep_params.keys():
+        var = sweep_params[key].pyomo_object
+        var_str = sweep_params[key].pyomo_object.name
+        var_str_list.append(var_str)
+        output_dict["sweep_params"][var_str] =  _create_component_output_skeleton(var, num_samples)
 
-        sweep_params: A dictionary containing the values to vary with the format
-                      ``sweep_params['Short/Pretty-print Name'] =
-                      (model.fs.variable_or_param[index], lower_limit, upper_limit, num_samples)``.
-                      A uniform number of samples ``num_samples`` will be take between
-                      the ``lower_limit`` and ``upper_limit``.
+    if outputs is None:
+        # No outputs are specified, so every Var, Expression, and Objective on the model should be saved
+        for pyo_obj in model.component_data_objects((pyo.Var, pyo.Expression, pyo.Objective), active=True):
+            # Only need to save this variable if it isn't one of the value in sweep_params
+            if pyo_obj.name not in var_str_list:
```

Comment:
I don't think this worth the effort. Why not just report out everything?

If we want to maintain the current behavior, `var_str_list` needs to be made into a `set`. Otherwise this for loop is quadratic in the number of `Var`s / `Constraint`s / `Expression`s.

Reply:
Or better still, we should just use a `ComponentSet` from `pyomo.common.collections`, which can just store the pyomo objects directly, as opposed to its string representation.

Reply:
I agree, that there is a better way of doing this but can we push it to a subsequent PR?

Reply:
Yes, being consistent about whether we primarily operate on strings/names or pyomo objects would be good. Currently, there's enough of a reliance on strings for creating dictionaries that the `var_str_list` as a `set` works well here, but we could revisit it in the future.